### DISCUSSION TITLE
Helix crashes when using "Tab" to cycle completion menu

### DISCUSSION THREAD

**Discussion by adambigg-s** [helix-editor/helix#11311] [Troubleshooting]
When using "Tab" or "Shift-Tab" to cycle through completion menus, Helix crashes with: 

```
thread 'main' panicked at helix-core\src\transaction.rs:483:9:
Positions [(301, Before)] are out of range for changeset len 0!
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Using "Ctrl-p" and "Ctrl-n" works fine, but I thought maybe this bug should be known. My Helix version is helix 24.7 (4c183550) built from source.

**Comment by adambigg-s** [comment]
Ok, I have been trying to diagnose the source of this issue. While I don't know what is causing this, I have some more info: 

1.  using "Ctrl-n" does cause a crash, and if I FIRST use "Ctrl-n" to navigate the completion menu, I can then revert to using "Tab" without a crash
2. navigating a completion menu with only one option does not result in a crash 
3. Using "Tab" to navigate into a menu with more than one option results in a crash 100% of the time, instantly
4. It has nothing to do with my config file, when running with config.toml completely commented out I am still getting this issue

Could this be caused by a faulty build? Should I try rebuilding? I never experienced this issue until today.

**Comment by the-mikedavis** [comment]
This looks to be the same as https://github.com/helix-editor/helix/issues/11291
